### ISSUE TITLE
Error: unable to load VPN connection editor

### ISSUE THREAD

**Issue by krzykorzeniewski** [linuxmint/cinnamon-control-center#319]
### Distribution

Mint 21.2

### Package version

Cinnamon 5.8.4

### Graphics hardware in use

NVIDIA Geforce RTX 3060Ti LHR

### Frequency

Always

### Bug description

When trying to establish VPN connection using Secure Socket Tunneling Protocol (SSTP) the following error message shows up in the "Add VPN" window: "Error: unable to load VPN connection editor". Basically when trying to establish connection using this protocol there is no option to fill Gateway, Username, password fields etc.

### Steps to reproduce

1. Install sstp-client: **sudo apt install sstp-client**, then use **sudo apt install network-manager-sstp** and reset the network manager: **sudo systemctl restart NetworkManager**
2. Go to Network, Add VPN, select Secure Socket Tunneling Protocol (SSTP) 

![Screenshot from 2023-11-11 11-36-33](https://github.com/linuxmint/cinnamon/assets/130040100/bb64943a-237a-4e94-a602-81a916e0083e)

### Expected behavior

Instead of the error message being shown, there should be option to fill in neccessary information such as Gateway or User Name

**Comment by claudiux** [comment]
@k0rz3ni3wsky  Once you've created your connection, use `nm-connection-editor` to edit it more precisely.
@leigh123linux Maybe add a direct access to `nm-connection-editor` from c-c-c?

**Comment by leigh123linux** [comment]
@claudiux  I think the issue is bigger than that, other network plugins fail in a similar way.

**Comment by claudiux** [comment]
@leigh123linux We're not going to rewrite the NetworkManager code, are we?

**Comment by leigh123linux** [comment]
> @leigh123linux We're not going to rewrite the NetworkManager code, are we?
We should consider rebasing the network code in the c-c-c network panel.

**Comment by claudiux** [comment]
@leigh123linux You're right. In particular, the different types of VPN are poorly managed. Anything else?

**Comment by ghisvail** [comment]
Same issue when setting up an OpenConnect VPN connection

**Comment by dkosovic** [comment]
I did the following modification to print out an error when trying to load a GTK3 VPN plug-in with cinnamon-control-center:
```patch
diff --git a/panels/network/connection-editor/ce-page-vpn.c b/panels/network/connection-editor/ce-page-vpn.c
index a2aaa5d..fc70538 100644
--- a/panels/network/connection-editor/ce-page-vpn.c
+++ b/panels/network/connection-editor/ce-page-vpn.c
@@ -93,10 +93,12 @@ load_vpn_plugin (CEPageVpn *page, NMConnection *connection)
 {
        CEPage *parent = CE_PAGE (page);
         GtkWidget *ui_widget, *failure;
+        GError *err = NULL;
 
         page->editor = nm_vpn_editor_plugin_get_editor (page->plugin,
                                                         connection,
-                                                        NULL);
+                                                        &err);
+
         ui_widget = NULL;
         if (page->editor)
                 ui_widget = GTK_WIDGET (nm_vpn_editor_get_widget (page->editor));
@@ -104,6 +106,8 @@ load_vpn_plugin (CEPageVpn *page, NMConnection *connection)
        if (!ui_widget) {
                g_clear_object (&page->editor);
                 page->plugin = NULL;
+                g_warning ("Could not load VPN editor plugin: %s", err ? err->message : "Unknown error");
+                g_error_free (err);
                return;
        }
         vpn_cinnamonify_editor (ui_widget);
```

The error turns out to be the following for GTK3 VPN plugins like sstp, l2tp and openvpn that are using `NMACertChooser` :

`(cinnamon-control-center:90815): WARNING **: 21:00:00.001: Could not load VPN editor plugin: .:181:1 Invalid object type 'NMACertChooser'`

There used to be a workaround in place for `NMACertChooser` but was removed with commit https://github.com/linuxmint/cinnamon-control-center/commit/0f4d212874c4fbee18b860963d0a5c7bd54dcfd1

**Comment by udorronsoro** [comment]
Hi:

Any news on when this will be fixed or if there's a temporary workaround?

Thank you!

**Comment by dkosovic** [comment]
Use `nm-connection-editor` was suggested earlier as a workaround to create/edit the VPN connections.

**Comment by dkosovic** [comment]
NetworkManager-openvpn is using NMACertChooser and has the following commit to workaround the NMACertChooser issue :
- https://gitlab.gnome.org/GNOME/NetworkManager-openvpn/-/commit/8cde95234832f10acd8f6a7c375354e948ee53c1

None of the other NetworkManager VPN plugins (fortisslvpn, sstp, l2tp, vpnc and openconnect) that use NMACertChooser have a similar workaround.

**EDIT:**
VPN editor plugins using NmaCertChooser or NMACertChooser bug reports or fixes :
- NetworkManager-l2tp >= 1.20.18 now include a fix for cinnamon-control-center compatibility:
   + https://github.com/nm-l2tp/NetworkManager-l2tp/commit/b9ba6a69fc4af37511780f7f935c144f2d2b2b9c
 - NetworkManager-sstp has a commit fix :
   + https://gitlab.gnome.org/GNOME/network-manager-sstp/-/commit/76fc8eb10cafb48d1e4985ac34a1bfa62980eb69
 - NetworkManager-openconnect merge request fix :
   + https://gitlab.gnome.org/GNOME/NetworkManager-openconnect/-/merge_requests/73
 - NetworkManager-vpnc bug report :
   + https://gitlab.gnome.org/GNOME/NetworkManager-vpnc/-/issues/15
- NetworkManager-fortisslvpn has a commit fix :
  + https://gitlab.gnome.org/GNOME/NetworkManager-fortisslvpn/-/commit/0296450f9bb8b3f34e0032103a9c5ba359553320

**Comment by TobHaeg** [comment]
> @k0rz3ni3wsky Once you've created your connection, use `nm-connection-editor` to edit it more precisely. @leigh123linux Maybe add a direct access to `nm-connection-editor` from c-c-c?

Hello, i was today also struggling with this issue. The german router manufacturer AVM is using this protocol in his Fritzbox.
I'm using linux mint with cinnamon.

It took me some time to find this solution here. As it seems to take a long time to solve the problem, pleas add a hint in the error message to nm-connection-editor in the error message.

**Comment by dkosovic** [comment]
I had hoped that updated VPN clients with fixes would have been released by now. Even with [CVE-2025-9615](https://github.com/advisories/GHSA-w24g-6mf8-65cj), several VPN clients that rely on NMACertChooser or NmaCertChooser still haven’t published new releases

I've submitted pull request https://github.com/linuxmint/cinnamon-control-center/pull/376 as a fix for loading NMACertChooser or NmaCertChooser based VPN clients with `cinnamon-control-center`.
